be/src/olap/rowset/segment_v2/column_zone_map.h

```diff
@@ -0,0 +1,99 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <vector>
+
+#include "common/status.h"
+#include "util/coding.h"
+#include "util/slice.h"
+#include "util/faststring.h"
+#include "olap/olap_cond.h"
+#include "gen_cpp/segment_v2.pb.h"
+
+namespace doris {
+
+namespace segment_v2 {
+
+// This class encode column pages' zone map.
+// The binary format is like that
+// Header | Content
+// Header: 
+//      number of elements (4 Bytes)
+// Content:
+//      array of page zone map
+class ColumnZoneMapBuilder {
```

Comment:
Because the BinaryPlainPageBuilder will limit the page size. I should modify the is_page_full.
OK, I will modify it to adapt to this situation. I think it can also be reused in the future.